### PR TITLE
Conform to PSR-4 and add composer support

### PR DESCRIPTION
**Pull request by mallardduck**
This is a pretty big looking PR - but it really just looks bigger than it is. This is a huge step towards the suggestions I provided in #3. Overall there really aren't that many changes in lines; supporting PSR-4 required a lot of file moves, subtle namespace changes and etc. Due to this the changes look much larger than they are in reality.

That said - the changes needed to support PSR-4 for easier composer support are breaking. This likely will necessitate a full Major version bump - assuming the package is meant to conform to SemVer.

**Note:** This should be merged in after #4 & #5 are resolved. Once those are in I can rebase these for merge. Also the Docs will need to be regenerated, however I'm unsure how to do this and don't see it documented.

## Biggest Changes

1. Classes now organized in a PSR-4 compliant manner.
2. Utility namespaced functions now exist as methods inside Utility class.
3. Adds a composer.json file and initial composer support.
4. Includes a non-composer based loader file. (Providing two routes to using the PHP SDK for Contentstack)

### CODE REVIEW

**Comment by mallardduck** [comment]
I'm unsure with how to do unit testing using the current test cases. As such I haven't touched those - since even if I modified them correctly I can't execute them. I'm still going to give updating them a shot in hopes I can figure out how tests should work.

**Comment by mallardduck** [comment]
Hey @ninadhatkar,

I figured out how the tests were meant to be run and have been able to give them a shot. However I did notice a handful of issues with the tests and began fixing them. After fixing one issue I discovered another one.

As I wasn't having luck with my branch I wanted to confirm Master branch had passing tests. Based on my testing I'm seeing that even on the master branch the unit tests are not 100% passing. Checking out master and running the tests yields the same two failures I found initially.

These are for: `EntriesTest::testErrorMethod` and `EntriesTest::testFindCount`.

Hopefully this will be acceptable. I'm sure that by the end of these changes we can get 100% passing tests. It will just take a bit more tweaking.

**Comment by mallardduck** [comment]
I could keep working on this, however I think it would benefit us if #5 was merged first and then #4 can be closed.

Once those other items have been settled it will be easier to figure out the right path forward. I could see going two ways: 1) we fix the tests to make them 100% functional on master, then rebase this off that, then continue the work; or, 2) we can have me fix the tests within this PR and they'll be 100% passing when this is merged in.

Personally I'm more partial to going with option 2 - it just seems like it will be more simple in the long run. Going this route is likely a more direct path towards a more modern PHP SDK library.

File: loader.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?php
+
+$base_folder = __DIR__;
+
+$composer = json_decode(file_get_contents("$base_folder/composer.json"), 1);
+
+if (!isset($composer['autoload'])) {
+  throw new \OutOfBoundsException('The autoload section is not defined in the compser.json file');
```
Comment (mcarcangiu-ea) [inline loader.php]: Just a typo ;)
```suggestion
  throw new \OutOfBoundsException('The autoload section is not defined in the composer.json file');
```

**Review by mcarcangiu-ea — APPROVED** [review]

**Comment by uttamukkoji** [comment]
Closing this request as our developers added PSR 4 and Composer support.

**Pull request closed without merging**